admin_guide/scheduling/scheduler.adoc

```diff
@@ -0,0 +1,819 @@
+
+
+Using `nodeSelector` in a pod configuration, you can ensure that pods are only
+placed onto nodes with specific labels.
+
+
+. Ensure you have the desired labels
+ifdef::openshift-enterprise,openshift-origin[]
+(see xref:../../admin_guide/manage_nodes.adoc#updating-labels-on-nodes[Updating
+Labels on Nodes] for details)
+endif::openshift-enterprise,openshift-origin[]
+ifdef::openshift-dedicated[]
+(request changes by opening a support case on the
+https://access.redhat.com/support/[Red Hat Customer Portal])
+endif::openshift-dedicated[]
+and xref:../../admin_guide/managing_projects.adoc#using-node-selectors[node selector]
+set up in your environment.
+//tag::node-selectors[]
++
+For example, make sure that your pod configuration features the `nodeSelector`
+value indicating the desired label:
++
+----
+apiVersion: v1
+kind: Pod
+spec:
+  nodeSelector:
+    <key>: <value>
+...
+----
+
+. Modify the master configuration file
+(*_/etc/origin/master/master-config.yaml_*) in two places:
++
+.. Add `nodeSelectorLabelBlacklist` to the `admissionConfig` section with
+the labels that are assigned to the node hosts you want to deny pod placement:
++
+====
+----
+...
+admissionConfig:
+  pluginConfig:
+    PodNodeConstraints:
+      configuration:
+        apiversion: v1
+        kind: PodNodeConstraintsConfig
+        nodeSelectorLabelBlacklist:
+          - kubernetes.io/hostname
+          - <label>
+...
+----
+====
+
+.. Then, add the same to the `kubernetesMasterConfig` section to restrict direct pod creation:
++
+====
+----
+...
+kubernetesMasterConfig:
+  admissionConfig:
+    pluginConfig:
+      PodNodeConstraints:
+        configuration:
+          apiVersion: v1
+          kind: PodNodeConstraintsConfig
+          nodeSelectorLabelBlacklist:
+            - kubernetes.io/hostname
+            - <label_1>
+...
+----
+====
+
+. Restart {product-title} for the changes to take effect.
+ifdef::openshift-origin[]
++
+----
+# systemctl restart origin-master
+----
+endif::[]
+ifdef::openshift-enterprise[]
++
+----
+# systemctl restart atomic-openshift-master
+----
+endif::[]
+
+== Control Pod Placement to Projects
+
+The _PodNodeSelector_ admission controller determines where a pod can be placed using xref:../../architecture/core_concepts/pods_and_services.adoc#labels[labels on projects] and node selectors specified in pods.
+
+The *PodNodeSelector* admission controller allows you to force pods onto nodes associated with a specific project and prevent pods from being scheduled in those nodes. 
+
+To schedule pods onto specific project nodes:
+
+. Activate the *PodNodeSelector* admission controller:
++
+.. Modify the master configuration file (/etc/origin/master/master-config.yaml) to add the admission controller plug-in and labels:
++
+----
+admissionConfig:
+  pluginConfig:
+    PodNodeSelector:
+      configuration:
+        apiversion: v1
+        kind:podNodeSelectorPluginConfig: <1>
+          clusterDefaultNodeSelector: region=west <2>
```

Comment:
Could you provide some more information about clusterDefaultNodeSelector? Something like that if a project does not have any node selectors specified, pods for that project will be merged with clusterDefaultNodeSelector.